packages/asc-ui/src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import { ReactComponent as Search } from '@datapunt/asc-assets/lib/Icons/Search.svg'
+import IconButton from '../IconButton'
+import { InputStyle } from '../../styles/components'
+import IconButtonStyle from '../../styles/components/IconButtonStyle'
+import SearchBarStyle from '../../styles/components/SearchBarStyle'
+import { KeyboardKeys } from '../../types'
+import Input from '../Input'
+
+type SearchBarProps = {
+  minWidth?: string
+  maxWidth?: string
+  padding?: string
+  styledComponent?: any
+  placeholder?: string
+  onTextChanged: Function
+  onSearch: Function
+}
+
+const SearchBar: React.FC<SearchBarProps> = ({
+  children,
+  styledComponent,
+  placeholder,
+  onTextChanged,
+  onSearch,
+  ...otherProps
+}) => {
+  const ExtendedSearchBarStyle = styledComponent
+
+  const [state, setState] = React.useState({
+    text: '',
+  })
+
+  const handleTextChanged = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setState({
+      text: e.target.value,
+    })
+    onTextChanged(state.text)
```

Comment:
Could you verify / write a test if you really get the updated state here? I had this issue once, where after setting the state it didn't immediately updated...

Reply:
will do